authenticate.js

```diff
@@ -0,0 +1,42 @@
+function jsonParse(value, cb) {
+	var err = null, result
+	try {
+		result = JSON.parse(value)
+	} catch (e) {
+		err = e
+	}
+	cb(err, result)
+}
+
+module.exports = function a(emitter, tokenDb, tokenLock) {
+	return function authenticate(token, cb) { //cb(err, credentials)
+		if (!cb) cb = function () {}
+
+		if (!token) {
+			setTimeout(cb, 0, new Error('No token found'))
+		} else {
+			tokenLock.readLock(function (unlock) {

```

Comment:
huh, I guess since there are only two locks, and they should both be write locks, this library may as well use mutexify directly.  Once [this pull request](https://github.com/mafintosh/mutexify/pull/2) is merged, anyway.

We shouldn't be locking the whole db though, we should be able to lock for just that key.